functions/src/callable/reports.ts

```diff
@@ -9,6 +9,46 @@ import {
     getCollectionDocs, QuizAttemptDocument
 } from "../helpers/database";
 
+interface CourseAttempt {
+    courseId: string;
+    endTime: Date;
+    pass: boolean | null;
+    startTime: Date;
```

Comment:
Note: Whenever working with Firestore, it has it's own Timestamp object (firestore.Timestamp, https://firebase.google.com/docs/reference/node/firebase.firestore.Timestamp)